org.eclipse.sisu.plexus/src/org/codehaus/plexus/component/configurator/converters/basic/PathConverter.java

```diff
@@ -0,0 +1,138 @@
+        {
+            Object result = super.fromConfiguration( lookup, configuration, type, enclosingType, loader, evaluator, listener );
+            if ( result instanceof String )
+            {
+                // this always assumes the default filesystem
+                Path path = getPath( FileSystems.getDefault(), (String) result );
+                if ( !path.isAbsolute() )
+                {
+                    // relative paths are are given relative to base directory (not necessarily the working directory)
+                    // therefore make them absolute via ExpressionEvaluator.alignToBaseDirectory(...)
+                    return evaluator.alignToBaseDirectory( path.toFile() ).toPath();
+                }
+                return path;
+            }
+            return result;
+        }
+        // complex configuration (containing filesystem URI, parameter, and path)
+        else
+        {
+            try
+            {
+                Object object = fromExpression( configuration.getChild( "filesystem-uri" ), evaluator, type );
+                final URI uri;
+                if ( object instanceof String )
+                {
+                    uri = new URI( (String) object );
+                }
+                else if ( object instanceof URI )
+                {
+                    uri = (URI) object;
+                }
+                else
+                {
+                    throw new ComponentConfigurationException( "Configuration element with name 'filesystem-uri' does not contain a URI" );
+                }
+                MapConverter mapConverter = new MapConverter();
+                object = mapConverter.fromConfiguration( lookup, configuration.getChild( "filesystem-properties" ), Map.class, enclosingType, loader, evaluator );
+                final Map<String,?> env;
+                if ( object instanceof Map )
+                {
+                    env = (Map<String, ?>) object;
+                }
+                else
+                {
+                    env = Collections.emptyMap();
+                }
+                object = fromExpression( configuration.getChild( "path" ), evaluator, type );
+                final String path;
+                if ( object instanceof String )
+                {
+                    path = (String) object;
+                }
+                else
+                {
+                    throw new ComponentConfigurationException( "Configuration element with name 'path' does not contain a String" );
+                }
+                FileSystem fileSystem;
+                try
+                {
+                    fileSystem = FileSystems.getFileSystem( uri );
+                }
+                catch ( FileSystemNotFoundException e )
+                {
+                    fileSystem = FileSystems.newFileSystem( uri, env );
+                    // TODO: where to close the FS?
```

Comment:
@cstamas Any hints where/how to close the new FileSystem? Is there a hook in Plexus one can easily leverage here?